crates/shared/src/price_estimation/cached.rs

```diff
@@ -0,0 +1,107 @@
+use super::{Estimate, PriceEstimating, PriceEstimationError, Query};
+use cached::{Cached, TimedSizedCache};
+use std::{sync::Mutex, time::Duration};
+
+/// Price estimator wrapper that caches Ok results for some time.
+pub struct CachingPriceEstimator {
+    inner: Box<dyn PriceEstimating>,
+    cache: Mutex<TimedSizedCache<Query, Estimate>>,
+}
+
+impl CachingPriceEstimator {
+    pub fn new(inner: Box<dyn PriceEstimating>, max_age: Duration, max_size: usize) -> Self {
+        Self {
+            inner,
+            cache: Mutex::new(TimedSizedCache::with_size_and_lifespan_and_refresh(
```

Comment:
If query is not updated after max_age, then cache.cache_get(query) returns None right?

Reply:
Yes, it will return None for too old entries.